jvm-packages/dev/build.sh

```diff
@@ -1,21 +1,37 @@
 #!/usr/bin/env bash
+#
+# Copyright (c) 2019 by Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+BASEDIR="$( cd "$( dirname "$0" )" && pwd )" # the directory of this file
 
-set -x
-
-export JAVA_HOME=/usr/lib/jvm/java-1.8.0
-export MAVEN_OPTS="-Xmx3000m"
-export CMAKE_CXX_COMPILER=/opt/rh/devtoolset-2/root/usr/bin/gcc
-export CXX=/opt/rh/devtoolset-2/root/usr/bin/g++
-export CC=/opt/rh/devtoolset-2/root/usr/bin/gcc
-
-export PATH=$CXX:$CC:/opt/rh/python27/root/usr/bin/python:$PATH
-
-scl enable devtoolset-2 bash
-scl enable python27 bash
-
-rm /usr/bin/python
-ln -s /opt/rh/python27/root/usr/bin/python /usr/bin/python
-
-# build xgboost
-cd /xgboost/jvm-packages;ulimit -c unlimited;mvn package
+while true; do
+  read -p  "Do you want rebuild and tag the docker image locally?" yn
+  case $yn in
+      [Yy]* ) docker build -t dmlc/xgboost4j-build "${BASEDIR}"; break;;
+      [Nn]* ) break;;
+      * ) echo "Invalid Response: please answer [y]es or [n]o";;
+  esac
+done
```

Comment:
let's replace all interactive questions with the passed in arguments

Reply:
I think it makes it less usable when quickly developing local changes. 

That is, you will have to type something like `--skip-build` every time you don't want to rebuild the Dockerfile. (This also means we need to write documentation about it.)

I will defer to your opinion however.

Reply:
Yes, by default, we should keep unit test and skip rebuild image

Reply:
Also, the image wont rebuild unless the Dockerfile has changed, so do we even need to have this flag?

The issue with having skip building on default, is that if you have never built it before, the `docker run` step will fail, because the image clearly doesn't exist on your local cache.

Reply:
Then why you have this interactive question?

Reply:
It originally forced a rebuild by clearing the cache. 

Do you want me to introduce that behaviour with flag to force the rebuild, and default behaviour to NOOP if no changes are detected in Dockerfile?

Reply:
I have removed this, it will now attempt to rebuild every time, and only do something if you change the Dockerfile, or have never built it locally before.